core/tests/ts-integration/src/utils.ts

```diff
@@ -175,6 +271,26 @@ export class NodeSpawner {
         await waitForNodeToStart(proc, options.apiWeb3JsonRpcHttpUrl);
         return new Node(proc, options.apiWeb3JsonRpcHttpUrl, NodeType.MAIN);
     }
+
+    public restoreConfig() {
+        if (this.generalConfigPath != void 0 && this.originalConfig != void 0)
+            fsSync.writeFileSync(this.generalConfigPath, this.originalConfig, 'utf8');
+    }
+
+    private readFileConfig() {
+        if (this.generalConfigPath == void 0)
+            throw new Error('Trying to set property in config while not in file mode');
+        const generalConfig = fsSync.readFileSync(this.generalConfigPath, 'utf8');
+        return YAML.parse(generalConfig);
+    }
+
+    private writeFileConfig(config: any) {
+        if (this.generalConfigPath == void 0)
+            throw new Error('Trying to set property in config while not in file mode');
+
+        const newGeneralConfig = YAML.stringify(config);
+        fsSync.writeFileSync(this.generalConfigPath, newGeneralConfig, 'utf8');
+    }
 }
```

Comment:
that is obvious, but a little bit tricky part, because js sometimes treat hex values as numbers and store later as number. and it could ruin the parsing logic

Reply:
I have checked that there are no unexpected changes after parsing and stringifying again:

```
  console.log
    Config differences:
     --- /tmp/original_config.yml       2024-10-23 18:06:48.290023034 +0200
    +++ /tmp/new_config.yml     2024-10-23 18:06:48.290023034 +0200
    @@ -60,7 +60,7 @@
     operations_manager:
       delay_interval: 100
     state_keeper:
    -  transaction_slots: 8192
    +  transaction_slots: 1
       block_commit_deadline_ms: 2500
       miniblock_commit_deadline_ms: 1000
       miniblock_seal_queue_capacity: 10
    @@ -157,6 +157,8 @@
         poll_period: 5
         num_samples_for_blob_base_fee_estimate: 10
         internal_pubdata_pricing_multiplier: 1
    +    internal_enforced_l1_gas_price: 2000000000
    +    internal_enforced_pubdata_price: 2000000000
       watcher:
         confirmations_for_eth_event: 0
         eth_node_poll_interval: 300
    @@ -324,19 +326,19 @@
       max_rows_to_dispatch: 100
       max_retries: 5
     base_token_adjuster:
    -  price_polling_interval_ms: 30000
    -  price_cache_update_interval_ms: 2000
    +  price_polling_interval_ms: 1000
    +  price_cache_update_interval_ms: 500
       max_tx_gas: 80000
       default_priority_fee_per_gas: 1000000
       max_acceptable_priority_fee_in_gwei: 100000000000
    -  l1_receipt_checking_sleep_ms: 30000
    +  l1_receipt_checking_sleep_ms: 1000
       l1_receipt_checking_max_attempts: 3
       l1_tx_sending_max_attempts: 3
    -  l1_tx_sending_sleep_ms: 30000
    +  l1_tx_sending_sleep_ms: 1000
       halt_on_error: false
       price_fetching_max_attempts: 3
       price_fetching_sleep_ms: 5000
    -  l1_update_deviation_percentage: 10
    +  l1_update_deviation_percentage: 0
     basic_witness_input_producer:
       db_path: /home/ert/repos/zksync-era-apu-base-token-integration/./chains/valcus/db/main/basic_witness_input_producer
       window_size: 3
    @@ -344,9 +346,10 @@
     external_price_api_client:
       source: forced
       client_timeout_ms: 10000
    -  forced_numerator: 314
    -  forced_denominator: 1000
    +  forced_numerator: 300
    +  forced_denominator: 100
       forced_next_value_fluctuation: 3
    +  forced_fluctuation: 20
     consensus:
       server_addr: 127.0.0.1:3254
       public_addr: 127.0.0.1:3254

      at NodeSpawner.writeFileConfig (src/utils.ts:312:29)
```